zerver/views/report.py

```diff
@@ -141,3 +144,30 @@ def report_error(request: HttpRequest, user_profile: UserProfile, message: Text=
     ))
 
     return json_success()
+
+@csrf_exempt
+def report_csp_violations(request: HttpRequest) -> HttpResponse:
+    if request.method != 'POST':
+        return json_error(_('Only post requests supported!'))
+
+    payload = ujson.loads(request.body)
```

Comment:
We can do this with `payload=REQ(argument_type='body')` in the arguments, I think.

Reply:
addressed!